docs/core/diagnostics/event-counters.md

```diff
@@ -0,0 +1,353 @@
+---
```

Comment:
this article as a whole has a lot of code of substantial size. I suggest moving this into a snippets project and referencing that code. This way we get CI on that code.